dbt_semantic_interfaces/parsing/where_filter/where_filter_dimension.py

```diff
@@ -32,6 +33,14 @@ def grain(self, time_granularity: str) -> QueryInterfaceDimension:
         self.time_granularity_name = time_granularity
         return self
 
+    def descending(self, _is_descending: bool) -> QueryInterfaceDimension:
+        """Set the sort order for order-by."""
+        raise InvalidQuerySyntax("descending is invalid in the where parameter and filter spec")
+
+    def date_part(self, _date_part: str) -> QueryInterfaceDimension:
+        """Date part to extract from the dimension."""
+        raise InvalidQuerySyntax("date_part isn't currently supported in the where parameter and filter spec")
+
```

Comment:
I saw this in the other implementations as well and I wonder if there's a way for us to structure the protocols so that things are just typed, but I don't know if that's worth the complexity.

Reply:
Related question: in practice, will anybody other than us (or another implementer of a MetricFlow query interface) ever initialize one of these things directly? I don't think the WhereFilterDimension is a thing any end user will ever interact with except via our parser, is that correct?

Reply:
> structure the protocols so that things are just typed

I believe this is typed correctly. Users of the JDBC interface don't get the benefits of an LSP, so they are reliant on error messages to help them write correct code.

> will anybody other than us (or another implementer of a MetricFlow query interface) ever initialize one of these things directly?

Right, I think this is just an internal class.